File: a5_n^2_sort/ex_tasks/a_insertion_sort.cpp
#include <iostream>

using namespace std;

void insertion_sort(int n, int* arr) {
	int i = 1;
	while (i < n) {
		int j = i;
		while (j > 0 && arr[j-1] > arr[j]) {
			swap(arr[j-1], arr[j]);
			--j;
		}
		++i;
	}
}

int main() {
	int n;
	cin >> n;
	int* arr = new int[n];
	for (int i = 0; i < n; ++i) {
		cin >> arr[i];
	}
	insertion_sort(n, arr);
	for (int i = 0; i < n; ++i) {
		cout << arr[i] << ' ';
	}
	cout << '\n';

	delete [] arr;
	return 0;
}

File: a5_n^2_sort/ex_tasks/a_insertion_sort.py
def insertion_sort(n: int, arr: list) -> None:
    i = 1
    while i < n:
        j = i
        while j > 0 and arr[j-1] > arr[j]:
            arr[j], arr[j-1] = arr[j-1], arr[j]
            j -= 1
        i += 1


n = int(input())
arr = list(map(int, input().split()))
insertion_sort(n, arr)
print(arr)